### PR TITLE
chore(evergreen): authenticate with docker so that we don't make unauthenticated pulls

### PR DESCRIPTION
**Pull request by lerouxb**
This should help with the rate limiting.

### CODE REVIEW

File: .evergreen/start-atlas-cloud-cluster.sh [hunk 1]
```diff
@@ -89,6 +89,9 @@ cleanup() {
 
 trap cleanup EXIT
 
+echo "Logging in to docker"
+echo "${DOCKERHUB_PASSWORD}" | docker login -u ${DOCKERHUB_USERNAME} --password-stdin
```
Comment (gribnoysup) [inline .evergreen/start-atlas-cloud-cluster.sh:93]: Not to block this, but this script is also supposed to be used locally to run these tests, what will happen if these are missing? Is it just silently fails? Does it mess with the login that docker desktop is requiring on macos?

File: .evergreen/print-compass-env.js [hunk 2]
```diff
@@ -113,6 +115,8 @@ function printCompassEnv() {
 
   // https://jira.mongodb.org/browse/NODE-6320
   printVar('GYP_DEFINES', `kerberos_use_rtld=${process.platform === 'linux'}`);
+
+  printVar('DOCKER_CONFIG', `${originalPWD}/.evergreen/docker-config`);
```
Comment (lerouxb) [inline .evergreen/print-compass-env.js:119]: Not exactly a secret but we already have this code running everywhere and we need to build an abs path somehow 🤷

File: .evergreen/print-compass-env.js [hunk 3]
```diff
@@ -74,6 +74,8 @@ function printCompassEnv() {
     pathsToPrepend.unshift('/opt/mongodbtoolchain/v4/bin');
   }
 
+  pathsToPrepend(`${originalPWD}/.evergreen/docker-config/bin`);
```
Comment (lerouxb) [inline .evergreen/print-compass-env.js:77]: No idea if this should use /cygdrive/c or /cygdrive/z.. we'll see.

File: .evergreen/print-compass-env.js (hunk 2 above)
Comment (mcasimir) [reply .evergreen/print-compass-env.js:119]: this should not be everywhere, but just where we use docker without arifactory, which we use for signatures

**Review by mcasimir — APPROVED** [review]

File: .evergreen/print-compass-env.js (hunk 2 above)
Comment (lerouxb) [reply .evergreen/print-compass-env.js:119]: That's gonna be tricky

Comment (lerouxb) [reply .evergreen/print-compass-env.js:119]: It seems to have logged in to artifactory using the existing username/password vars we have for that and worked just fine.

We currently have most of our env vars together and this way it makes use of our existing code to that makes sure we don't accidentally leak secrets in logs. This way it should also be impossible to forget to log in and get all of evergreen rate limited if we ever start using docker in more places.

So I think we should just keep an eye on it for now and we can make more PRs to isolate that var if we have to. Worst case we just temporarily revert this PR.

**Pull request merged**